src/python/bot/fuzzers/honggfuzz/engine.py

```diff
@@ -0,0 +1,169 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""honggfuzz engine interface."""
+
+import glob
+import os
+import re
+
+from bot.fuzzers import dictionary_manager
+from bot.fuzzers import engine
+from system import environment
+from system import new_process
+
+_DEFAULT_ARGUMENTS = [
+    '-n',
+    '1',  # single threaded
+    '--exit_upon_crash',
+    '-v',  # output to stderr
+    '-z',  # use clang instrumentation
+    '-P',  # persistent mode
+    '-S',  # enable sanitizers
+    '--rlimit_rss',
+    '2048',
+    '--timeout',
+    '25',
+]
+
+_CRASH_REGEX = re.compile('Crash: saved as \'(.*)\'')
+_HF_SANITIZER_LOG_PREFIX = 'HF.sanitizer.log'
+
+
+def _get_runner():
+  """Get the honggfuzz runner."""
+  honggfuzz_path = os.path.join(environment.get_value('BUILD_DIR'), 'honggfuzz')
+  return new_process.ProcessRunner(honggfuzz_path)
+
+
+def _find_sanitizer_stacktrace(reproducers_dir):
+  """Find the sanitizer stacktrace from the reproducers dir."""
+  for stacktrace_path in glob.glob(
+      os.path.join(reproducers_dir, _HF_SANITIZER_LOG_PREFIX + '*')):
+    with open(stacktrace_path, 'rb') as f:
+      return f.read()
+
+  return None
+
+
+class HonggfuzzEngine(engine.Engine):
+  """honggfuzz engine implementation."""
+
+  @property
+  def name(self):
+    return 'honggfuzz'
+
+  def prepare(self, corpus_dir, target_path, build_dir):
+    """Prepare for a fuzzing session, by generating options. Returns a
+    FuzzOptions object.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+
+    Returns:
+      A FuzzOptions object.
+    """
+    arguments = []
+    dict_path = dictionary_manager.get_default_dictionary_path(target_path)
+    if os.path.exists(dict_path):
+      arguments.extend(['--dict', dict_path])
+
+    return engine.FuzzOptions(corpus_dir, arguments, [])
+
+  def fuzz(self, target_path, options, reproducers_dir, max_time):
+    """Run a fuzz session.
+
+    Args:
+      target_path: Path to the target.
+      options: The FuzzOptions object returned by prepare().
+      reproducers_dir: The directory to put reproducers in when crashes
+          are found.
+      max_time: Maximum allowed time for the fuzzing to run.
+
+   Returns:
+      A FuzzResult object.
+    """
+    runner = _get_runner()
+    arguments = _DEFAULT_ARGUMENTS[:]
+    arguments.extend(options.arguments)
+    arguments.extend([
+        '-f',
```

Comment:
nit: does it have longer, more readable option, similar to --run_time ?

Reply:
Yes, done!